#! /bin/sh /usr/share/dpatch/dpatch-run
## drop_unnecessary_dependencies.dpatch by  <till.kamppeter@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Do not link libcups.so and libcupsimage.so against unnecessary libraries.
## DP: This avoids unnecessary package dependencies for both libcups, as well
## DP: as for packages using cups-config (Debian #438067).

@DPATCH@
diff -urNad cups-1.4.0~svn8773~/Makedefs.in cups-1.4.0~svn8773/Makedefs.in
--- cups-1.4.0~svn8773~/Makedefs.in	2009-08-23 12:16:58.000000000 +0200
+++ cups-1.4.0~svn8773/Makedefs.in	2009-08-23 18:12:14.000000000 +0200
@@ -132,7 +132,7 @@
 DBUS_NOTIFIERLIBS =	@DBUS_NOTIFIERLIBS@
 DNSSD_BACKEND	=	@DNSSD_BACKEND@
 DSOFLAGS	=	-L../cups @DSOFLAGS@
-DSOLIBS		=	@DSOLIBS@ $(COMMONLIBS)
+DSOLIBS		=	@DSOLIBS@
 DNSSDLIBS	=	@DNSSDLIBS@
 FONTS		=	@FONTS@
 IMGLIBS		=	@IMGLIBS@
@@ -144,9 +144,9 @@
 LEGACY_BACKENDS	=	@LEGACY_BACKENDS@
 LIBCUPSORDER	=	@LIBCUPSORDER@
 LIBCUPSIMAGEORDER =	@LIBCUPSIMAGEORDER@
-LINKCUPS	=	@LINKCUPS@ $(SSLLIBS) $(DNSSDLIBS)
+LINKCUPS	=	@LINKCUPS@ $(DNSSDLIBS)
 LINKCUPSIMAGE	=	@LINKCUPSIMAGE@
-LIBS		=	$(LINKCUPS) $(COMMONLIBS)
+LIBS		=	$(LINKCUPS) $(SSLLIBS) $(COMMONLIBS)
 OPTIM		=	@OPTIM@
 OPTIONS		=
 PAMLIBS		=	@PAMLIBS@
diff -urNad cups-1.4.0~svn8773~/cups/Makefile cups-1.4.0~svn8773/cups/Makefile
--- cups-1.4.0~svn8773~/cups/Makefile	2009-08-23 12:16:16.000000000 +0200
+++ cups-1.4.0~svn8773/cups/Makefile	2009-08-23 18:21:18.000000000 +0200
@@ -291,7 +291,7 @@
 libcups.so.2 libcups.sl.2:	$(LIBOBJS)
 	echo Linking $@...
 	$(DSO) $(ARCHFLAGS) $(DSOFLAGS) -o $@ $(LIBOBJS) $(LIBGSSAPI) \
-		$(SSLLIBS) $(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
+		$(SSLLIBS) $(DNSSDLIBS) -lpthread $(LIBZ)
 	$(RM) `basename $@ .2`
 	$(LN) $@ `basename $@ .2`
 
@@ -304,7 +304,7 @@
 	echo Linking 32-bit $@...
 	-mkdir 32bit
 	$(DSO) $(ARCH32FLAGS) $(DSO32FLAGS) -o $@ $(LIB32OBJS) $(LIBGSSAPI) \
-		$(DNSSDLIBS) $(SSLLIBS) $(COMMONLIBS) $(LIBZ)
+		$(DNSSDLIBS) $(SSLLIBS) -lpthread $(LIBZ)
 	$(RM) 32bit/libcups.so
 	$(LN) libcups.so.2 32bit/libcups.so
 
@@ -317,7 +317,7 @@
 	echo Linking 64-bit $@...
 	-mkdir 64bit
 	$(DSO) $(ARCH64FLAGS) $(DSO64FLAGS) -o $@ $(LIB64OBJS) $(LIBGSSAPI) \
-		$(DNSSDLIBS) $(SSLLIBS) $(COMMONLIBS) $(LIBZ)
+		$(DNSSDLIBS) $(SSLLIBS) -lpthread $(LIBZ)
 	$(RM) 64bit/libcups.so
 	$(LN) libcups.so.2 64bit/libcups.so
 
@@ -335,7 +335,7 @@
 		-exported_symbols_list libcups.exp \
 		-sectorder __TEXT __text $(LIBCUPSORDER) \
 		$(LIBOBJS) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
-		$(COMMONLIBS) $(LIBZ)
+		-lpthread $(LIBZ)
 	$(RM) libcups.dylib
 	$(LN) $@ libcups.dylib
 
@@ -348,7 +348,7 @@
 	echo Creating $@...
 	$(DSO) $(DSOFLAGS) -Wl,-bexport:libcups_s.exp -o libcups_s.o \
 		$(LIBOBJS) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
-		$(COMMONLIBS) $(LIBZ) -lm
+		-lpthread $(LIBZ)
 	$(RM) $@
 	$(AR) $(ARFLAGS) $@ libcups_s.o
 
@@ -361,7 +361,7 @@
 	echo Linking $@...
 	$(CC) $(ARCHFLAGS) $(DSOFLAGS) -o $@ $(LIBOBJS:.o=.lo) \
 		-rpath $(LIBDIR) -version-info 2:8 $(LIBGSSAPI) $(SSLLIBS) \
-		$(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
+		$(DNSSDLIBS) -lpthread $(LIBZ)
 
 
 #
